### PR TITLE
feat(git): submodule-aware git diff — protocol v1.1 + group-by-repo UI

### PR DESCRIPTION
**Pull request by anur4ag**
## Summary

Surfaces changes an agent makes **inside a git submodule** (e.g. `traycer/` in `traycer-internal`) in the Git Diff panel. The panel was single-repo by construction and previously showed only an opaque `Subproject commit` gitlink row for a dirty submodule.

This is the **client + protocol half** of a coordinated two-PR change. The host half (git discovery/classification/resolvers) lives in `traycer-internal`; this PR ships the wire contract and the UI.

## Protocol (`@traycer/protocol`)

- Freeze the stream's file schema as `gitChangedFileV10Schema`. `git.subscribeStatus` stays **frozen at v1.0** — stream methods can't be minor-versioned safely, so all new data is unary-only and the broadcaster projects results back through the frozen v1.0 schema before emitting.
- Minor-bump **only `git.listChangedFiles`** to **v1.1** with a `v1.0 → v1.1` upgrade path:
  - request: additive `includeSubmodules` (default `false`), so lightweight callers keep the cheap parent-only snapshot;
  - response: parent `files` gain a nullable `gitlink` pointer descriptor, plus a host-composed `submodules[]` nested snapshot (default `[]`).
- The gitlink pointer is a minimal discriminated union: `normal` (recorded pin, submodule HEAD, a plain `diverged` pin≠HEAD boolean, and the porcelain `c`/`m`/`u` dirty flags) or `conflicted` (base/ours/theirs pins). **Working-tree only** — no ahead/behind relation computation and no commits-ahead file listing.
- Each `submodules[]` section carries the submodule's own working-tree files, `branch`/`repoState`, the pointer descriptor, and an `availability` state (`ok` | `unavailable` + reason) so a submodule the host can't inspect renders as a **visible degrade, never a silent omission**.
- `git.getFileDiff` / `git.getFileDiffs` stay **v1.0-only**: submodule files are diffed stage-based by pointing `runningDir` at the submodule repo root, so the v1.0 request/response shapes are unchanged.

## GUI (`gui-app`)

- Renders the host-composed nested snapshot **grouped by repository**: a repo tree with the parent plus one section per submodule, a demoted "Submodule reference" pointer row on the parent, and explicit degraded / old-host / submodule-unavailable states.
- Routes submodule working-tree diffs by `repoRoot` through the existing v1.0 diff RPCs.
- A bounded 5s refresh keeps already-dirty submodule sections fresh — inner edits don't move the parent gitlink row, so the v1.0 stream's change token alone can't catch them.

## Backward compatibility

Additive minor bump. New GUI + old host → `submodules: []` / `gitlink: null` (parent-only view); old GUI + new host → new fields stripped by minor-skew projection. No handshake break; the released method-name set is frozen by `released-surface-compat.test.ts`.

## Verification

- Per-package `compile` + `lint` clean; protocol + gui-app suites green — registry validation, transport skew in both directions, stream no-leak on snapshot + updated frames, composition/rendering coverage.
- Cold-reviewed per-ticket. Full `nx affected` workspace checks run in CI.

## Notes

- Paired with the `traycer-internal` host PR (traycer-internal#4202). The internal gitlink pin bump lands as a separate follow-up after this merges.

### CODE REVIEW

File: clients/gui-app/src/components/epic-canvas/git-diff/submodule-reference-row.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            data-testid={`submodule-reference-refresh-${view.parentPath}`}
+          >
+            <RotateCcw
+              className={cn("size-3", props.isRefreshing && "animate-spin")}
+              aria-hidden
+            />
+            Refresh
+          </button>
+        </div>
+      ) : null}
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/submodule-reference-row.tsx]: Verified against the codebase — this is the app's established refresh-affordance idiom, not an ad-hoc loading spinner.

The spinning `RotateCcw` (`animate-spin`) driven by `useRefreshSpinner` is the shared pattern for refresh buttons across `gui-app` (`refresh-icon-button.tsx`, `git-diff-panel-actions.tsx`, `diff-tab-toolbar.tsx`, `empty-states/no-changes-in-worktree.tsx`, `chat-accumulated-changes-panel.tsx`, and more). `submodule-unavailable.tsx` uses the same `useRefreshSpinner` hook.

`AgentSpinningDots` renders braille **loading dots** (`⠋⠙⠹…`) — a loading-state indicator, not a rotating refresh icon. The AGENTS.md guideline is explicitly scoped to *"when this component can represent the loading state"*, which a refresh button's spinning refresh icon is not. Swapping it for dots would be inconsistent with every other refresh button in the app.

Not actionable — resolving.

File: clients/gui-app/src/components/epic-canvas/git-diff/repo-tree.tsx [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+            type="button"
+            role="treeitem"
+            aria-level={row.kind === "root" ? 1 : 2}
+            aria-selected={row.selected}
+            aria-expanded={
+              row.kind === "root" && row.hasChildren ? true : undefined
+            }
+            aria-disabled={isDisabled ? true : undefined}
+            disabled={isDisabled}
+            tabIndex={index === tabbableIndex ? 0 : -1}
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/repo-tree.tsx]: Confirmed valid against current code — `repo-tree.tsx:218` still sets native `disabled={isDisabled}`.

The roving-tabindex navigation moves focus via `rowRefs.current[i]?.focus()`, but a native-`disabled` `<button>` cannot receive focus (`.focus()` is a no-op). So when a disabled root sits between two selectable roots, sequential ArrowUp/Down can't move onto it — and because it never becomes the active element, the next Arrow keydown recomputes the same disabled target and stays stuck, so the user can't traverse *past* it either. Every non-tabbable row has `tabIndex={-1}`, so Tab can't reach them as a fallback.

Fix: drop the native `disabled` and rely on `aria-disabled` + the existing `activateRow` guard (`if (target.disabledLabel !== null) return;`), which already blocks Enter/Space/click/ArrowRight activation. The row stays focusable but non-activatable. The existing "blocks activation" test stays green — click now dispatches `onClick`, but the guard blocks `onSelectRoot`.

Routing this fix to the author.

File: clients/gui-app/src/hooks/git/use-git-list-changed-files-with-submodules.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+ * needs a timer to pick up further inner edits. 5s matches the host's own poll
+ * class (UX-confirmed freshness for this review surface).
+ */
+const SUBMODULE_BOUNDED_REFRESH_MS = 5000;
+
+export interface GitListChangedFilesWithSubmodulesResult {
+  readonly data: GitListChangedFilesResponseV11 | null;
+  readonly isPending: boolean;
+  readonly error: HostRpcError | null;
+}
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/hooks/git/use-git-list-changed-files-with-submodules.ts:24]: Partly confirmed — the deviation is deliberate, but the in-code rationale is imprecise.

The doc says it rolls a bespoke `useQuery` "so `client` stays out of the cache key." That isn't the actual differentiator: `useHostQuery` also keeps `client` out of the key — it keys on `readiness.hostId` + method/params and takes an arbitrary `client` argument, so the worktree-scoped host routing (`useHostClientFor(entry)`) could be passed straight into it.

The real reasons this can't fold into `useHostQuery`:
1. It needs the bespoke, co-owned key `gitQueryKeys.listChangedFilesWithSubmodules(...)` that the subscription/refresh pair and `invalidateGitSubmoduleSnapshot` target. `useHostQuery` forces the `["host", hostId, method, params]` key shape.
2. The `changeToken`-driven `refetch` effect + bounded `refetchInterval` freshness machinery.

The narrowing to `{ data, isPending, error }` is a separate, intentional choice (the two consumers only read those three), but it does deviate from the "return raw `UseQueryResult`" guideline.

So: legitimate exception, but worth (a) tightening the doc comment to the real reason (co-owned query key + freshness), and (b) consciously confirming the narrowed surface. Leaving open for the author.

File: clients/gui-app/src/components/epic-canvas/git-diff/repo-tree.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/repo-tree.tsx]: Fixed in `25c39d8`. The native `disabled={isDisabled}` was removed from the tree-row `<button>` (`repo-tree.tsx`); `aria-disabled` stays and the `activateRow` guard (`if (target.disabledLabel !== null) return;`) keeps disabled roots non-activatable. Disabled roots are now focusable, so roving-tabindex ArrowUp/Down can move onto and past them without activating them. Resolving.

File: clients/gui-app/src/stores/home/landing-draft-store.ts [hunk 4]
```diff
@@ -195,16 +196,14 @@ function isLandingDraftDocContent(value: unknown): value is JsonContent {
...
 }
 
-function parseLandingDraftSelection(
-  value: DesktopJsonValue | null,
-): DraftSelection | null {
+function parseLandingDraftSelection(value: unknown): DraftSelection | null {
   if (!isRecord(value)) return null;
   const { from, to } = value;
   if (typeof from !== "number" || typeof to !== "number") return null;
   return { from, to };
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/stores/home/landing-draft-store.ts:210]: Fixed in `25c39d8`. `parseLandingDraftSelection` now rejects non-finite `from`/`to` (`!Number.isFinite(from) || !Number.isFinite(to)`), matching the `Number.isFinite` check already used by `parseLandingDraftLastTouchedAt`. Resolving.

File: clients/gui-app/src/components/epic-canvas/git-diff/repo-tree.tsx [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+/** Index of the root row that owns the submodule row at `index` (else itself). */
+function parentRootIndex(
+  visibleRows: ReadonlyArray<VisibleRow>,
+  index: number,
+): number {
+  for (let i = index - 1; i >= 0; i--) {
+    if (visibleRows[i].kind === "root") return i;
+  }
+  return index;
+}
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/repo-tree.tsx]: Applied in `6140878`. `parentRootIndex` now uses `visibleRows.slice(0, index).findLastIndex((r) => r.kind === "root")` and returns `index` when none is found — behavior-equivalent to the old reverse loop. Resolving.

File: clients/gui-app/src/components/epic-canvas/git-diff/repo-tree.tsx [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+        rows.push({
+          kind: "submodule",
+          key: `sub:${node.repoRoot}`,
+          node,
+          selected: selected.repoRoot === node.repoRoot,
+        });
+      }
+    }
+    return rows;
+  }, [roots, selected, activeRootSubmodules]);
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/repo-tree.tsx]: Applied in `6140878`. `visibleRows` is now built with `roots.flatMap(...)` returning `[rootRow, ...submoduleRows]`, preserving identical keys, order, `selected`/`hasChildren` logic, and active-root-only expansion. Resolving.

File: clients/gui-app/src/components/epic-canvas/git-diff/git-diff-panel-body-live.tsx [hunk 7]
```diff
@@ -62,300 +53,304 @@ export function GitDiffPanelBodyLive(
...
-      if (!isGitSelectable(row)) continue;
+    for (const row of gitRows) {
       void prefetch({
         hostId: row.hostId,
         runningDir: row.runningDir,
         ignoreWhitespace,
       });
     }
-  }, [ignoreWhitespace, prefetch, rows]);
+  }, [ignoreWhitespace, prefetch, gitRows]);
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/git-diff-panel-body-live.tsx:196]: Applied in `6140878`. The prefetch effect now uses `gitRows.forEach((row) => { … })` instead of the `for…of` loop. Resolving.

File: clients/gui-app/src/components/epic-canvas/git-diff/selected-repo-changes.tsx [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+          <RepoStateBanner
+            state={source.repoState}
+            repoMode={source.repoMode}
+            conflictCount={conflictCount}
+          />
+        ) : null}
+        <NoChangesInWorktree lastUpdatedAtMs={subscription.pollStartedAtMs} />
+      </div>
+    );
+  }
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/selected-repo-changes.tsx]: Applied in `6140878`. The `RepoStateBanner` conditional is hoisted into a single `const banner` computed once and reused in both the empty-state early return and the main return. Resolving.

File: clients/gui-app/src/components/epic-canvas/git-diff/__tests__/repo-tree.test.tsx [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+  repoRoot: "/repo",
+};
+
+function renderTree(props: {
+  roots: ReadonlyArray<RepoTreeRootRow>;
+  selected: GitPanelSelectedRepo;
+  activeRootSubmodules: ReadonlyArray<RepoTreeSubmoduleNode>;
+  onSelectRoot?: (row: WorktreeBindingSelectorRow) => void;
+  onSelectSubmodule?: (node: RepoTreeSubmoduleNode) => void;
+}) {
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/__tests__/repo-tree.test.tsx]: Applied in `6140878`. The `renderTree` helper's `onSelectRoot` / `onSelectSubmodule` props are now required (no `?:`), and every call site passes `vi.fn()` explicitly. Resolving.

File: clients/gui-app/src/components/epic-canvas/git-diff/__tests__/repo-tree.test.tsx [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+      selected: rootSelection,
+      activeRootSubmodules: [],
+      onSelectRoot,
+    });
+    const disabledRoot = screen.getByTestId("git-repo-tree-root-not-git");
+    expect(disabledRoot.getAttribute("aria-disabled")).toBe("true");
+    expect(disabledRoot.textContent).toContain("not git");
+    fireEvent.click(disabledRoot);
+    expect(onSelectRoot).not.toHaveBeenCalled();
+  });
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/__tests__/repo-tree.test.tsx]: Applied in `6140878`. Added a keyboard-nav regression test covering a disabled root between two selectable roots: ArrowDown lands focus onto the disabled root and then past it to the next selectable root, and ArrowUp reverses. This guards the a11y focus-trap fix. Resolving.

File: clients/gui-app/src/components/epic-canvas/git-diff/empty-states/submodule-unavailable.tsx [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+      >
+        <RotateCcw
+          className={cn(
+            "mr-1.5 size-3.5",
+            refresh.refreshing && "animate-spin",
+          )}
+          aria-hidden
+        />
+        Refresh
+      </Button>
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/empty-states/submodule-unavailable.tsx:62]: Intentionally **not** deduped — after review these are genuinely different affordances, not a copy-paste:

- `submodule-unavailable.tsx`: a centered shadcn `Button` (`variant="outline" size="sm"`) empty-state action driven by `useRefreshSpinner`, icon `size-3.5`.
- `submodule-reference-row.tsx`: a raw inline `<button>` inside a warning banner, driven by `props.isRefreshing`, icon `size-3`.

They differ in primitive, sizing, spin-state source, and container. A shared extraction would have to parametrize all of that and would worsen both call sites rather than simplify them. Leaving as-is (open for the author's final call).

File: clients/gui-app/src/components/epic-canvas/git-diff/submodule-reference-row.tsx [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+            "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring",
+          )}
+        >
+          {content}
+        </button>
+      ) : (
+        <div className="flex w-full min-w-0 items-center gap-1.5 px-1 py-0.5 text-ui-sm">
+          {content}
+        </div>
+      )}
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/submodule-reference-row.tsx]: Intentionally **deferred**. A behavior-preserving discriminated union can't also enforce the "conflicted ⇒ no `repoRoot`" invariant: `repoRoot` is derived from whether a matching submodule changeset is present, independently of conflict state. Forcing `repoRoot: null` on the conflicted variant would flip the row from an interactive `<button>` to a non-interactive `<div>` *if the host ever emits a matching changeset for a conflicted path* — a rendering change, not just a type change — and it ripples into `buildSubmoduleReferenceRows` + the component. Skipped to avoid that rendering risk; worth revisiting if/when the host contract is tightened to guarantee conflicted paths carry no changeset. Leaving open for the author.

**Comment by anur4ag** [comment]
@coderabbitai full review

**Comment by anur4ag** [comment]
@coderabbitai full review

File: clients/gui-app/src/components/epic-canvas/git-diff/file-sections.tsx [hunk 13]
```diff
@@ -25,6 +27,9 @@ export interface FileSectionsProps {
   readonly visibleFiles: ReadonlyArray<GitChangedFile>;
   readonly pathRangesByPath: ReadonlyMap<string, HighlightRanges>;
   readonly forceExpanded: boolean;
+  readonly hideEmptySections: boolean;
+  readonly sectionCollapseController: GitDiffSectionCollapseController | null;
+  readonly virtualized: boolean;
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/file-sections.tsx:32]: Triaged and leaving unchanged. The `FileSections` / `FileTree` duplication is a broader structural extraction with no current behavior bug, and this PR is already review-fix scoped around submodule-aware Git Diff behavior. A new abstraction here would add churn without improving correctness. Validation after the accepted fixes: submodule `pre-commit run --all-files` passed. Resolving as not actionable for this PR.

File: clients/gui-app/src/components/epic-canvas/git-diff/git-changed-files-view.tsx [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+        files={props.files}
+        query={appliedQuery}
+        onClearQuery={handleClear}
+        hideEmptySections={false}
+        sectionCollapseController={null}
+        virtualized
+      />
+    </div>
+  );
+}
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/git-changed-files-view.tsx:119]: Fixed in `0d0055a`. Added coverage around the changed-files view interactions, including debounced search behavior and clearing via the input clear control / Escape path so the search UI behavior is locked down. Validation: focused Git Diff tests passed and submodule `pre-commit run --all-files` passed. Resolving.

File: clients/gui-app/src/components/epic-canvas/git-diff/git-changed-files-view.tsx [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+    (event: KeyboardEvent<HTMLInputElement>) => {
+      if (event.key !== "Escape") return;
+      event.preventDefault();
+      handleClear();
+      event.currentTarget.blur();
+    },
+    [handleClear],
+  );
+
+  useEffect(() => clearPendingDebounce, [clearPendingDebounce]);
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/git-changed-files-view.tsx:74]: Triaged and leaving unchanged. Extracting the debounce behavior into a shared hook is low-value right now: there are only a couple of local call sites and the current code is explicit and covered by the interaction tests added in `0d0055a`. Validation: submodule `pre-commit run --all-files` passed. Resolving as not worth changing in this PR.

File: clients/gui-app/src/lib/git/git-diff-repo-switcher.ts [hunk 16]
```diff
@@ -0,0 +1,306 @@
...
+  if (!row.isGitRepo) return "not git";
+  return formatWorktreeFolderDisabledReason(row);
+}
+
+function searchableText(parts: ReadonlyArray<string | null>): string {
+  return parts
+    .filter((part): part is string => part !== null && part.length > 0)
+    .join(" ")
+    .toLowerCase();
+}
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/lib/git/git-diff-repo-switcher.ts:79]: Triaged and leaving unchanged. The duplicate `searchableText` derivation is small, local, and not currently drifting; extracting it would be a readability tradeoff rather than a correctness fix. Validation after accepted changes: focused Git Diff tests passed and submodule `pre-commit run --all-files` passed. Resolving as low-value for this PR.

File: clients/gui-app/src/lib/git/git-diff-repo-switcher.ts [hunk 17]
```diff
@@ -0,0 +1,306 @@
...
+      rootLabel: label,
+    }),
+  );
+  const submoduleStateChanged = submodules.some(
+    (node) => node.hasChanges || node.unavailable,
+  );
+  const selectedRoot =
+    selected !== null &&
+    selected.hostId === input.row.hostId &&
+    selected.rootRunningDir === input.row.runningDir;
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/lib/git/git-diff-repo-switcher.ts:163]: Triaged and leaving unchanged. The active-root checks are intentionally close to their branch-specific behavior, and the existing coverage exercises the switcher behavior. A helper would not reduce meaningful risk in this review-fix pass. Validation: focused Git Diff tests passed and submodule `pre-commit run --all-files` passed. Resolving.

File: clients/gui-app/src/components/epic-canvas/git-diff/empty-states/git-roots-unavailable.tsx [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+            "mr-1.5 size-3.5",
+            refresh.refreshing && "animate-spin",
+          )}
+          aria-hidden
+        />
+        Retry
+      </Button>
+    </div>
+  );
+}
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/empty-states/git-roots-unavailable.tsx:69]: Triaged and leaving unchanged. The empty-state components share a visual pattern, but their copy and affordance context differ enough that extracting a shared component would mostly parameterize text and layout. No behavior issue found. Validation after accepted fixes: submodule `pre-commit run --all-files` passed. Resolving as low-value for this PR.

File: clients/gui-app/src/components/epic-canvas/git-diff/git-changed-file-row.tsx [hunk 19]
```diff
@@ -57,6 +55,73 @@ function RowStats(props: RowStatsProps): ReactNode {
...
+    >
+      <span className="min-w-0 truncate">
+        <HighlightedText text={parts.stem} ranges={stemRanges} />
+      </span>
+      <span className="shrink-0">
+        <HighlightedText text={parts.extension} ranges={extensionRanges} />
+      </span>
+    </span>
+  );
+}
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/components/epic-canvas/git-diff/git-changed-file-row.tsx:123]: Fixed in `0d0055a`. Added filename split/highlight regression coverage for the edge cases around path/name rendering so the row display behavior is protected. Validation: focused `git-changed-file-row` tests passed and submodule `pre-commit run --all-files` passed. Resolving.

File: clients/gui-app/src/lib/git/git-diff-repo-switcher.ts [hunk 20]
```diff
@@ -0,0 +1,306 @@
...
+      openTarget: {
+        workspacePath: selected.rootRunningDir,
+        hostId: selected.hostId,
+      },
+    };
+  }
+
+  const row = rows.find(
+    (item) => item.key === `root:${worktreeRowKey(root.row)}`,
+  );
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/lib/git/git-diff-repo-switcher.ts:270]: Triaged and leaving unchanged. A `rootRowKey` helper would centralize a tiny string composition, but the current uses are clear and covered by switcher behavior tests. The extraction is not worth the churn in this PR. Validation: focused Git Diff tests passed and submodule `pre-commit run --all-files` passed. Resolving as low-value.

File: clients/gui-app/src/hooks/git/use-git-list-changed-files-with-submodules.ts [hunk 21]
```diff
@@ -0,0 +1,176 @@
...
+ *
+ * Manual refresh is a plain `invalidateQueries` on this slot (see
+ * `GitDiffPanelActions`); there is no `refreshRelations` flag (the host no longer
+ * carries a relation cache to bypass).
+ *
+ * Rolls a bespoke `useQuery` against `client.request` (rather than
+ * `useHostQuery`) so `client` stays out of the cache key - it is transport
+ * identity, not data identity - mirroring the subscription/refresh pair that
+ * co-own the v1.0 slot.
+ */
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/hooks/git/use-git-list-changed-files-with-submodules.ts:78]: Triaged and leaving unchanged. The hook comment can be tightened later, but it is documentation-only and not on the changed code path for the accepted fixes. No behavior risk found. Validation after accepted changes: submodule `pre-commit run --all-files` passed. Resolving as low-value for this PR.

File: clients/gui-app/src/lib/git/git-repo-tree.ts [hunk 22]
```diff
@@ -0,0 +1,468 @@
...
+    if (file.gitlink === null) {
+      ordinaryFiles.push(file);
+      continue;
+    }
+    if (seenGitlinkPaths.has(file.path)) continue;
+    seenGitlinkPaths.add(file.path);
+    gitlinkFiles.push(file);
+  }
+  return { ordinaryFiles, gitlinkFiles };
+}
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/lib/git/git-repo-tree.ts:80]: Triaged and leaving unchanged. The current loop in `splitParentFiles` is simple, preserves first-gitlink-per-path ordering directly, and is already covered by tests. Rewriting it into filters with a side-effecting `Set` would not improve correctness. Validation: focused Git Diff tests passed and submodule `pre-commit run --all-files` passed. Resolving as not actionable.

File: clients/gui-app/src/lib/git/git-repo-tree.ts [hunk 23]
```diff
@@ -0,0 +1,468 @@
...
+    parentPath: changeset.parentPath,
+    // The submodule's parent-repo path is the label - the full path (not the
+    // basename) so nested or duplicate-basename submodules stay unambiguous.
+    label: changeset.parentPath,
+    headLabel: formatRepoHeadLabel(changeset.branch, headSha),
+    changeCount,
+    hasChanges: changeCount > 0 || hasPointerChange,
+    unavailable,
+  };
+}
```
[the remark this answers is not in the document]
Comment (anur4ag) [reply clients/gui-app/src/lib/git/git-repo-tree.ts:233]: Triaged and leaving unchanged. The duplicated head label derivation is small and close to each caller; extracting a helper would not remove meaningful complexity or fix a behavior issue. Existing tests cover the visible repo-tree output. Validation after accepted changes: submodule `pre-commit run --all-files` passed. Resolving as low-value for this PR.

**Comment by anur4ag** [comment]
@coderabbitai approve
